internal/functions/efgs/upload-batch.go

```diff
@@ -182,45 +185,81 @@ func uploadBatch(ctx context.Context, batch *efgsapi.DiagnosisKeyBatch, config *
 			return nil, err
 		}
 
-		logger.Debug("Some keys in batch were invalid or duplicated: %+v", parsedResponse)
+		logger.Debugf("Some keys in batch were invalid or duplicated: %+v", parsedResponse)
 		return &parsedResponse, nil
 	default:
 		logger.Debug("Batch upload failed. No key was uploaded")
 		return &parsedResponse, fmt.Errorf("HTTP %v: %v", res.StatusCode, string(body))
 	}
 }
 
-func handleErrorUploadResponse(ctx context.Context, resp *efgsapi.UploadBatchResponse, uploadConfig *uploadConfig, keys []*efgsapi.DiagnosisKeyWrapper) error {
+func handleErrorUploadResponse(ctx context.Context, resp *efgsapi.UploadBatchResponse, uploadConfig *uploadConfig, batchDbKeys []*efgsapi.DiagnosisKeyWrapper, diagnosesKeys []*efgsapi.DiagnosisKey) error {
 	logger := logging.FromContext(ctx).Named("efgs.handleErrorUploadResponse")
 
+	// The magic in this method is needed because EFGS uses weird sorting and we need to map indexes reported by EFGS (indexes in uplod batch)
+	// to index resp. key in the original batch. I'd love to implement this some less mind-blowing way but in a language where one has to
+	// implement a function for getting all keys/values of map, this way is just a smaller pain.
+
+	// This is needed for binary-search below
+	sort.Slice(batchDbKeys, func(i, j int) bool {
+		return bytes.Compare(batchDbKeys[i].KeyData, batchDbKeys[j].KeyData) > 0
+	})
+
+	// Find DB entity relevant to DiagnosisKey with given index
+	findRelevantKey := func(index int) *efgsapi.DiagnosisKeyWrapper {
+		lookFor := diagnosesKeys[index].KeyData
+
+		// This does binary search. The method is usually named accordingly in normal languages.
+		// Thanks God there exists the bytes.Compare method.
```

Comment:
how about we try to document it in a bit less passive aggressive way? ;)

Reply:
How about next time try to program in something more usable? :innocent: